metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/DataEvolutionUtils.java

```diff
@@ -517,6 +355,81 @@ private static boolean pluginHasSuccessfulRecords(
     return new ImmutablePair<>(previousPlugin, previousExecution);
   }
 
+  /**
+   * <p>
+   * This method returns a sequence of published harvest increments. This is a list of successive
+   * published harvests that must start with the most recent full published harvest and then
+   * includes all incremental published harvests (if any) that came after that.
+   * </p>
+   * <p>
+   * Note that a published harvest is a harvest that resulted in a index to publish (i.e. any
+   * harvest that is the root predecessor of an index to publish).
+   * </p>
+   *
+   * @param datasetId The dataset ID for which to obtain the chain.
+   * @return The chain, in the form of plugin-execution pairs that are ordered chronologically. Is
+   * never null, but can be empty if no such chain exists (i.e. the dataset does not have a
+   * published harvest or we find an index after the last full harvest that is invalid or did
+   * somehow not originate from a harvest).
+   */
+  public List<PluginWithExecutionId<? extends ExecutablePlugin<?>>> getPublishedHarvestIncrements(
```

Comment:
Don't we need a unit test for this?

Reply:
Sure, we could. I'll see if I can add one within reasonable time.